File: codeeval/easy/004sumofprimes/solution.c
#include <stdio.h>
#include <math.h>

int isprime(int n) {
    if (!(n&1)) return 0;
    for (int i = 3; i <= (int) sqrt(n); i += 2) {
        if (n % i == 0)
            return 0;
    }
    return 1;
}

const int primes(int n) {
    int sum = 2;
    for (int i = 3; n > 1; i += 2) {
        if (isprime(i)) {
            sum += i;
            --n;
        }
    }
    return sum;
}

int main(int argc, char *argv[]) {
    printf("%i\n", primes(1000));
    return 0;
}

File: codeeval/easy/004sumofprimes/solution.py
#!/usr/bin/env python

from sys import argv
from math import sqrt

def isprime(n):
    if n & 1 == 0:
        return False
    for i in range(3, int(sqrt(n))+1, 2):
        if n % i == 0:
            return False
    return True

def primes(n):
    sum = 2
    x = 3
    for i in range(n-1):
        sum += x
        x += 2
        while not isprime(x):
            x += 2
    return sum

print(primes(1000))
